modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java

```diff
@@ -604,4 +612,278 @@ private void deleteApplication(String applicationId) throws Exception {
+    }
+
+    @Test(dependsOnMethods = "createOrganizationsForPaginationTests")
+    public void testGetPaginatedOrganizationsWithDefaultLimit() {
+
+        // Test case 1: URL with LIMIT_QUERY_PARAM but no value.
+        String endpointURLWithEmptyLimit =
+                ORGANIZATION_MANAGEMENT_API_BASE_PATH + QUESTION_MARK + LIMIT_QUERY_PARAM + EQUAL + AMPERSAND +
+                        RECURSIVE_QUERY_PARAM + EQUAL + FALSE;
+
+        Response responseWithEmptyLimit = getResponseOfGetWithOAuth2(endpointURLWithEmptyLimit, m2mToken);
+
+        validateHttpStatusCode(responseWithEmptyLimit, HttpStatus.SC_OK);
+
+        validateOrganizationsForDefaultLimit(responseWithEmptyLimit, NUM_OF_ORGANIZATIONS_FOR_PAGINATION_TESTS);
+
+        // Test case 2: URL without LIMIT_QUERY_PARAM.
+        String endpointURLWithoutLimit =
+                ORGANIZATION_MANAGEMENT_API_BASE_PATH + QUESTION_MARK + RECURSIVE_QUERY_PARAM + EQUAL + FALSE;
+
+        Response responseWithoutLimit = getResponseOfGetWithOAuth2(endpointURLWithoutLimit, m2mToken);
+
+        validateHttpStatusCode(responseWithoutLimit, HttpStatus.SC_OK);
+
+        validateOrganizationsForDefaultLimit(responseWithoutLimit, NUM_OF_ORGANIZATIONS_FOR_PAGINATION_TESTS);
+    }
+
+    private List<Map<String, String>> createOrganizations(int numberOfOrganizations) throws JSONException {
+
+        List<Map<String, String>> newOrganizations = new ArrayList<>();
+
+        for (int i = 0; i < numberOfOrganizations; i++) {
+            JSONObject body = new JSONObject()
+                    .put(ORGANIZATION_NAME, String.format(ORGANIZATION_NAME_FORMAT, i));
+
+            Response response =
+                    getResponseOfPostWithOAuth2(ORGANIZATION_MANAGEMENT_API_BASE_PATH, body.toString(), m2mToken);
+
+            if (response.getStatusCode() == HttpStatus.SC_CREATED) {
+                // Extract the organization ID (UUID) from the response body.
+                JSONObject responseBody = new JSONObject(response.getBody().asString());
+                String organizationId = responseBody.getString("id");
+
+                // Store the created organization details.
+                Map<String, String> org = new HashMap<>();
+                org.put(ORGANIZATION_NAME, String.format(ORGANIZATION_NAME_FORMAT, i));
+                org.put(ORGANIZATION_ID, organizationId);
+                newOrganizations.add(org);
+            } else {
+                throw new RuntimeException("Failed to create organization " + i);
+            }
+        }
+
+        return newOrganizations;
+    }
+
+    private String getLink(List<Map<String, String>> links, String rel) {
+
+        for (Map<String, String> link : links) {
+            if (rel.equals(link.get(REL))) {
+                String href = link.get(HREF);
+                if (href.contains(AFTER_QUERY_PARAM + EQUAL)) {
+                    return href.substring(href.indexOf(AFTER_QUERY_PARAM + EQUAL) + 6);
```

Comment:
shall we use AFTER_QUERY_PARAM.length() instead of 6. (Assuming AFTER_QUERY_PARAM equals "after?"). Check the before param as well

Reply:
Done